finality-aleph/src/validator_network/manager/direction.rs

```diff
@@ -0,0 +1,225 @@
+use std::collections::{HashMap, HashSet};
+
+use aleph_primitives::AuthorityId;
+
+use crate::validator_network::Data;
+
+pub struct DirectedPeers<A: Data> {
+    own_id: AuthorityId,
+    outgoing: HashMap<AuthorityId, Vec<A>>,
+    incoming: HashSet<AuthorityId>,
+}
+
+fn bit_xor_sum_parity((a, b): (u8, u8)) -> u8 {
+    let mut result = 0;
+    for i in 0..8 {
+        result += ((a >> i) ^ (b >> i)) % 2;
+    }
+    result % 2
+}
+
+// Whether we shold call the remote or the other way around. We xor the peer ids and based on the
+// parity of the sum of bits of the result decide whether the caller should be the smaller or
+// greated lexicographically. They are never equal, because cryptography.
```

Comment:
Typo fixed, the number of slashes was on purpose (this is not a public function so it will not generate docs anyway), but I guess it doesn't hurt, so also switched.